apps/src/templates/studioHomepages/JoinSection.jsx

```diff
@@ -96,20 +104,49 @@ export default class JoinSection extends React.Component {
 
   handleKeyUp = event => {
     if (event.key === 'Enter') {
-      this.joinSection();
+      this.validateRecaptcha();
     } else if (event.key === 'Escape') {
-      this.setState(INITIAL_STATE);
+      this.setState({sectionCode: '', displayCaptcha: false});
     }
   };
 
-  joinSection = () => {
+  validateRecaptcha = () => {
```

Comment:
I think this function needs to be renamed because it joins the section if no sectionAttempts < 3?

Reply:
Any suggestions for a name?